simplify and fix complete_payment helper and refactor tests

dobertRowneySr opened the pull request:
Addresses:


* #3688 
* #3657 

#### Changes
* implemented royalty fee accounting proposed in #3688 
* `complete_payment` helper function cleaned up
* add test for case described in the security report with proper refactoring

#### Notes
* rebased on `carthage-substrate-v3`


┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202476311049904) by [Unito](https://www.unito.io)

L3pereira committed:
introduce_channel_state_bloat_bond_#3511

requested changes

Carthage T: Trait fix

L3pereira committed:
introduce_channel_state_bloat_bond_#3511

L3pereira committed:
Introduce video state bloat bond #3394

Introduce video state bloat bond #3394

dobertRowneySr committed:
test fixes

dobertRowneySr committed:
fix tests

dobertRowneySr committed:
fix tests

dobertRowneySr committed:
add revenue_split_rate to token data

dobertRowneySr committed:
add revenue_split_rate to token data

dobertRowneySr committed:
rebase

L3pereira committed:
remove_council_blog_pallet_and_proposals_#3662

remove_blog_proposals_#3662

remove_council_blog_pallet_and_proposals_#3662

remove_council_blog_typescript_pallet_and_proposals_#3662

remove_council_blog_typescript_pallet_and_proposals_#3662

requested changes + cargo fmt in types.rs

requests + types + metadata

chain metadata

dobertRowneySr committed:
remove categories from content

dobertRowneySr committed:
fix runtime

dobertRowneySr committed:
fix chainspec config

dobertRowneySr committed:
generate types library and metadata

L3pereira committed:
video state bloat bond Carthage T: Trait fix

dobertRowneySr committed:
simplify and fix complete_payment helper and refactor tests

dobertRowneySr committed:
aremove channel caterogy id from Genesis Config

dobertRowneySr committed:
fix clippy

dobertRowneySr committed:
fix clippy

dobertRowneySr committed:
generate types

dobertRowneySr committed:
update types and metadata

dobertRowneySr committed:
set next_sale_id starting value to 0

dobertRowneySr committed:
set next_sale_id starting value to 0

dobertRowneySr committed:
add test with failing destination member

Lezek123 commented on runtime-modules/content/src/tests/mock.rs:
```diff
@@ -536,7 +536,59 @@ impl Default for ExtBuilder {
+            platform_fee_percentage: self.platform_fee_percentage,
+            auction_starts_at_max_delta: self.auction_starts_at_max_delta,
+            max_auction_whitelist_length: self.max_auction_whitelist_length,
+        }
```
Probably worth refactoring this into a separate method

dobertRowneySr replied:
see 2228e1ed689475a41645a46f0da104a98894cdaf it's similar to the project token pallet now

dobertRowneySr committed:
rename allocation_amount -> revenue_amount
So the terminology is:
- allocation amount = what creator allocates for a revenue split
- revenue amount = what is actually provided to shareholders

Lezek123 commented on runtime-modules/content/src/tests/nft/buy_nft.rs:
```diff
@@ -4,62 +4,70 @@ use crate::tests::mock::*;
+            assert_eq!(
+                channel_reward_account_balance(1u64),
+                DEFAULT_NFT_PRICE - platform_fee,
+            );
```
You're only checking one account, but the original issue involved 2 accounts.
I suggest adding a check of `SECOND_MEMBER_ACCOUNT_ID` (- nft_price) and `COLLABORATOR_MEMBER_ID` (no balance change) also

Lezek123 reviewed (commented):
Looks good, just added 2 minor suggestions

Lezek123 commented:
One more thing - please run `cargo fmt`

dobertRowneySr committed:
disallow issuing a token with 0 revenue split rate

dobertRowneySr committed:
add test for lefover revenue split allocation amount

dobertRowneySr committed:
generate metadata & types

dobertRowneySr committed:
yarn format

L3pereira committed:
types generation and channel duplication fix

types generation

types generation and channel duplication fix

L3pereira committed:
types generation

dobertRowneySr committed:
swap names allocation <-> revenue

dobertRowneySr committed:
swap revenue <-> allocation in tests & traits

dobertRowneySr committed:
update chain meta data

dobertRowneySr committed:
send fund to controller account after rev. split

dobertRowneySr committed:
add test for leftover funds

dobertRowneySr committed:
fix naming inconsistencies

dobertRowneySr committed:
refactor build genesis config

dobertRowneySr committed:
add extra checks on actor balances for edge case

dobertRowneySr committed:
add extra checks on normal case with royalty

dobertRowneySr committed:
cargo format

dobertRowneySr committed:
fix naming inconsistency 2

dobertRowneySr committed:
cargo format

Lezek123 reviewed (commented):
LGTM, but the metadata and types need to be updated

dobertRowneySr committed:
fix clippy

dobertRowneySr committed:
generate metadata

dobertRowneySr committed:
Merge branch 'carthage' into audit3__fix_nft_payments

dobertRowneySr committed:
Merge branch 'carthage' into audit3__fix_nft_payments

dobertRowneySr committed:
generate metadata

Lezek123 reviewed (approved):
LGTM
